src/redux/actions/loginActions.js

```diff
@@ -60,11 +60,22 @@ export const validationResponse = (payload, type = INPUT_VALIDATION_FAILURE) =>
   payload,
 });
 
-export const validateCredentials = ({ username, password }) => dispatch => new Promise((resolve) => {
+export const validateCredentials = (
+  { username, password },
+) => dispatch => new Promise((resolve) => {
```

Comment:
`new Promise`? I'm very interested in why we felt we have to explicitly create a promise in this way?